core/modules/layout/layout.admin.inc

```diff
@@ -2471,18 +2483,24 @@ function layout_menu_item_form($form, &$form_state, LayoutMenuItem $menu_item) {
       '#name' => 'conditions[' . $access_key . '][configure]',
     );
   }
-  $form['conditions']['add'] = array(
-    '#type' => 'submit',
-    '#value' => t('Add visibility condition'),
-    '#attributes' => array('class' => array('layout-link-button', 'layout-access-add')),
-    '#submit' => array(
-      'layout_menu_item_form_condition_add'
+  $form['conditions']['links'] = array(
+    '#theme' => 'layout_action_links',
+    '#items' => array(
+      'add' => array(
```

Comment:
I think by adding these elements as a property (`#items`), this will stop the recursive scanning of the form for child elements, and behavior-properties like `#ajax` and `#submit` won't behave properly.

Reply:
Okay, I'll try with them as top-level children then.

Reply:
So perhaps add the children like this instead:

```
$form['conditions']['links']['add'] = array(
```

Then in your theme function loop over the form children keys and render each one.

Reply:
Yep, that fixed it!